omod/src/test/webapp/resources/scripts/navigatorHandlers.js

```diff
@@ -724,12 +725,15 @@ describe("Tests for simple form navigation handlers", function() {
                 firstQuestion.isSelected=true;
                 field.isSelected = true;
                 firstQuestion.fields = [field];
+                spyOn(secondField, 'select');
+                spyOn(secondField, 'toggleSelection');
+                spyOn(secondField, 'isValid').and.returnValue(false);
+                secondQuestion.fields = [ secondField ];
 
                 clickedQuestionHandler(questions, thirdQuestion, event);
 
                 expect(firstQuestion.onExit).not.toHaveBeenCalled();
                 expect(event.stopPropagation).toHaveBeenCalled();
-                expect(field.select).toHaveBeenCalled();
```

Comment:
I would has guessed that the following would be the expected behavior in this case... do any of these lines pass? (ie, the use clicks on the third question, but the second it invalid, so the first question is toggled and existed, and the section question and field gets selected) 

`expect(event.stopPropagation).toHaveBeenCalled();
                expect(firstQuestion.onExit).toHaveBeenCalled();
                expect(firstQuestion.toggleSelection).toHaveBeenCalled();
                expect(secondQuestion.toggleSelection).toHaveBeenCalled();
expect(secondField.select).toHaveBeenCalled()`